addons/survey/models/survey_survey.py

```diff
@@ -1170,6 +1170,7 @@ def action_end_session(self):
         if not self.env.user.has_group('survey.group_survey_user'):
             raise AccessError(_('Only survey users can manage sessions.'))
 
+        self.ensure_one()
         self.sudo().write({'session_state': False})
         self.user_input_ids.sudo().write({'state': 'done'})
         self.env['bus.bus']._sendone(self.access_token, 'end_session', {})
```

Comment:
This makes it explicit but is already the case implicitly, since the `write` or the `self.access_token` would crash if we have respectively 0 or more than 1 record